app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -807,9 +819,15 @@ public function getShippingAndTaxEstimate( $quote )
                 $label = $label . ' - ' . $rate->getMethodTitle();
             }
 
+            $rateCode = $rate->getCode();
+
+            if(empty($rateCode)) {
+                Mage::helper('boltpay/bugsnag')->notifyException(new Exception('Rate code is empty. ' . var_export($rate->debug(), true)));
+            }
+
             $option = array(
                 "service"   => $label,
-                "reference" => $rate->getCarrier() . '_' . $rate->getMethod(),
+                "reference" => $rateCode,
```

Comment:
This seems functional change, not logging change. Maybe this fixes the issue? 
The reference field is currently used later at 
> $quote->getShippingAddress()->setShippingMethod($referenceShipmentMethod)->save();

Reply:
I couldn't find any examples where the rate code wasn't the same as the carrier and method name, so I swapped with this since the rate code is actually what Magento expects as far as I can tell.